### PR TITLE
Implement LevelCube

### PR DESCRIPTION
**Pull request by CursedFlames**

### CODE REVIEW

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/level/chunk/MixinImposterProtoChunk.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import io.github.opencubicchunks.cubicchunks.world.level.chunklike.ImposterProtoClo;
+import io.github.opencubicchunks.cubicchunks.world.level.chunklike.LevelClo;
+import net.minecraft.world.level.chunk.ImposterProtoChunk;
+import net.minecraft.world.level.chunk.LevelChunk;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+@Mixin(ImposterProtoChunk.class)
+public abstract class MixinImposterProtoChunk implements ImposterProtoClo {
+    public LevelClo cc_getWrappedClo() {
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/level/chunk/MixinImposterProtoChunk.java]: is this supposed to be `@Override`?

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/LevelCube.java [hunk 2]
```diff
@@ -1,10 +1,437 @@
...
+                    this.unsaved = true;
+                    return previousState;
+                }
+            }
+        }
+    }
+
+    // Unused in vanilla as well
+    @Deprecated
+    @Override public void addEntity(Entity p_62826_) {}
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/LevelCube.java]: Thoughts on this being dasm just for consistency?

File: src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/LevelCube.java [hunk 3]
```diff
@@ -1,10 +1,437 @@
...
+    @TransformFrom(copyFrom = @CopyFrom(clazz = LevelChunk.class), value = "removeBlockEntityTicker(Lnet/minecraft/core/BlockPos;)V")
+    private native void removeBlockEntityTicker(BlockPos p_156413_);
+
+    @TransformFrom(copyFrom = @CopyFrom(clazz = LevelChunk.class), value = "runPostLoad()V")
+    public native void runPostLoad();
+
+    // Mojang why
+    public boolean isEmpty() {
+        return false;
+    }
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/world/level/cube/LevelCube.java]: perhaps also dasm?

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/level/cube/MixinLevelCube.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+// Needed for DASM to apply
+@Mixin(LevelCube.class)
+public abstract class MixinLevelCube extends MixinCubeAccess {
+    @Dynamic @Redirect(method = "cc_dasm$getBlockState", at = @At(value = "INVOKE", target = "Lio/github/opencubicchunks/cubicchunks/world/level/cube/LevelCube;getSectionIndex(I)I"))
+    private int cc_onGetBlockState_SectionIndex(LevelCube instance, int i, BlockPos pos) {
+        return Coords.blockToIndex(pos);
+    }
+
+    @Dynamic @Redirect(method = "cc_dasm$getFluidState(III)Lnet/minecraft/world/level/material/FluidState;", at = @At(value = "INVOKE", target = "Lio/github/opencubicchunks/cubicchunks"
+        + "/world/level/cube/LevelCube;getSectionIndex(I)I"))
```
Comment (NotStirred) [inline src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/level/cube/MixinLevelCube.java]: missing comments?
> All non-trivial mixins _**must**_ have a comment explaining their purpose

**Review by NotStirred — CHANGES_REQUESTED** [review]
A minor nit and some discussion points

File: src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/level/chunk/MixinImposterProtoChunk.java (hunk 1 above)
Comment (CursedFlames) [reply src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/world/level/chunk/MixinImposterProtoChunk.java]: yes.

**Review by NotStirred — APPROVED** [review]
lgtm assuming ci pass

**Pull request merged**